linode/manager
# M3-5431 EU Model Contract agreement banner + notification

## Description

This adds two things:
1.  Banner indicating the user must accept the EU Model Contract agreement
2. Handling for the EU Model Contract agreement notification

The new modal component sits in the GlobalNotifications wrapper component, and context is used in order to control the component from across the app. This pattern is debatable and honestly I didn’t think about it as much as I would like since we’re moving quickly. If you have other suggestions I’d love to consider them.

I also did a slight refactor of the DismissibleBanner component to be able to use the functionality from another component.

**TODO**: ~Listen for the correct API notification (I just mocked it in for now)~


## How to test

I included a mock notification in serverHandlers so if you use the MSW you should see the banner/notification.

To test dismissibility you'll need to turn the MSW off and remove `!hasComplianceNotification` from L23 of ComplianceBanner.tsx.

To test end-to-end you'll need to use devenv... (haven't done this myself yet).
